src/libpst.c

```diff
@@ -3564,8 +3622,13 @@ static int pst_getBlockOffsetPointer(pst_file *pf, pst_id2_tree *i2_head, pst_su
         }
     }
     else {
+        DEBUG_WARN(("Found internal %#x value.\n", offset));
         // internal index reference
         size_t subindex  = offset >> 16;
+        if (pf->do_read64 == 2) {
+            // Shift over 3 more bits for new flags.
+            subindex = subindex >> 3;
```

Comment:
I think this is the appropriate change - however, I think this value is always 0 in the example OST files I have, since everything works even without this line.  I am also still confused about how this implementation matches up with the documentation (https://msdn.microsoft.com/en-us/library/ff386694(v=office.12).aspx).  For example, I don't know why `pst_getBlockOffset` does:

```
of1 = offset >> 4;
```

instead of

```
of1 = (offset >> 5) - 1;
```

but there's clearly something funky going on with the offset calculation (since e.g. we then add 2 to this offset everywhere it's actually used as an index in `buf`).  This works, and changing it breaks, so...